specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -46,6 +47,48 @@ type MappedBusinessRuleDefs = {
 };
 
 export const businessRuleDefs: MappedBusinessRuleDefs = {
+  Address: {
+    customInit: (address) => {
+      if (address.isNew()) {
+        const setPrimary = (): void => {
+          address.set('isPrimary', true);
+          if (address.collection !== undefined) {
+            address.collection.models.forEach(
+              (other: SpecifyResource<Address>) => {
+                if (other.cid !== address.cid) other.set('isPrimary', false);
+              }
+            );
+          }
+        };
+        address.on('add', setPrimary);
+      }
+    },
+    fieldChecks: {
+      isPrimary: async (address): Promise<BusinessRuleResult> => {
+        if (
+          address.get('isPrimary') === true &&
+          (address.collection !== null || address.collection !== undefined)
+        ) {
+          address.collection.models.forEach(
```

Comment:
```suggestion
        )
          address.collection?.models.forEach(
```

Reply:
Fixed in https://github.com/specify/specify7/pull/4443/commits/a184d4cf817fd9fc0e0245bb9d1e29a9a28e36ec. 
I have also made the typing of a resource's collection more accurate and have amended the uses of `resource.collection` throughout the code base